### PR TITLE
better error handling in the STS tests

### PR DESCRIPTION
**Pull request by yuvalif**
aslo, give more accurate instruction on how to run the tests

Signed-off-by: Yuval Lifshitz <ylifshit@redhat.com>

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -60,6 +60,14 @@ You can run only the boto3 tests with::
 
 This section contains some basic tests for the AssumeRole, GetSessionToken and AssumeRoleWithWebIdentity API's. The test file is located under ``s3tests_boto3/functional``.
 
+To run the STS tests, the vstart cluster should ne started with the following parameter (in addition to any parameters already used with it)::
+
+        vstart.sh -o rgw_sts_key=abcdefghijklmnop -o rgw_s3_auth_use_sts=true
```
Comment (pritha-srivastava) [inline README.rst:65]: actually rgw_sts_key can be set to anything that is 128 bits in length.

**Comment by pritha-srivastava** [comment]
The changes look fine to me.

File: README.rst (hunk 1 above)
Comment (yuvalif) [reply README.rst:65]: but it does work with the above, right?
any reason to change that?

Comment (pritha-srivastava) [reply README.rst:65]: it works, we could just mention that the key given is an example or something similar.

**Comment by yuvalif** [comment]
> The changes look fine to me.

thanks. can you please approve?

**Comment by yuvalif** [comment]
@cbodley what is the qa/teuthology process for s3test changes?

File: README.rst (hunk 1 above)
Comment (pritha-srivastava) [reply README.rst:57]: @yuvalif : can you just add here that the key used is an example.

Comment (yuvalif) [reply README.rst:57]: added. see line 59

Comment (pritha-srivastava) [reply README.rst:57]: ok

File: README.rst [hunk 2]
```diff
@@ -52,6 +52,15 @@ You can run only the boto3 tests with::
 
 This section contains some basic tests for the AssumeRole, GetSessionToken and AssumeRoleWithWebIdentity API's. The test file is located under ``s3tests_boto3/functional``.
 
+To run the STS tests, the vstart cluster should ne started with the following parameter (in addition to any parameters already used with it)::
```
Comment (pritha-srivastava) [inline README.rst:55]: 'should be' and not 'should ne'

**Review by pritha-srivastava — APPROVED** [review]

**Comment by yuvalif** [comment]
> @cbodley what is the qa/teuthology process for s3test changes?

thanks for the info. documented it here: https://gist.github.com/yuvalif/0cb8cddb53bcd9a948af4e762eedb85d

**Comment by yuvalif** [comment]
teuthology results: http://pulpito.front.sepia.ceph.com/yuvalif-2023-02-16_16:32:18-rgw-main-distro-default-smithi/
has 5 failures a subset of the failures here: https://github.com/ceph/ceph/pull/48709.

multisite test seems to crash, but no core/backtrace is found

**Comment by cbodley** [comment]
cherry-picked to ceph-master

**Pull request merged**